Tools/ci-run.sh

```diff
@@ -34,7 +32,28 @@ elif [[ $OSTYPE == "darwin"* ]]; then
   export CC="clang -Wno-deprecated-declarations"
   export CXX="clang++ -stdlib=libc++ -Wno-deprecated-declarations"
 else
-  echo "No setup specified for $OSTYPE"
+  echo "Skipping compiler setup: No setup specified for $OSTYPE"
+fi
+
+if [[ $COVERAGE == "1" ]]; then
+  echo "Skip setting up compilation caches"
+elif [[ $OSTYPE == "msys" ]]; then
+  echo "Set up sccache"
+  echo "TODO: Make a soft symlink to sccache"
+else
+  echo "Set up ccache"
+  echo "Make a soft symlinks to ccache"
+
+  # sudo /usr/sbin/update-ccache-symlinks ?
+  cp ccache /usr/local/bin/
+  ln -s ccache /usr/local/bin/gcc
+  ln -s ccache /usr/local/bin/g++
+  ln -s ccache /usr/local/bin/cc
+  ln -s ccache /usr/local/bin/c++
+  ln -s ccache /usr/local/bin/clang
+  ln -s ccache /usr/local/bin/clang++
+
+  echo "/usr/lib/ccache" >> $GITHUB_PATH  # export ccache to path
```

Comment:
export ccache to path before using it
```suggestion
  echo "/usr/lib/ccache" >> $GITHUB_PATH  # export ccache to path

  cp ccache /usr/local/bin/
  ln -s ccache /usr/local/bin/gcc
  ln -s ccache /usr/local/bin/g++
  ln -s ccache /usr/local/bin/cc
  ln -s ccache /usr/local/bin/c++
  ln -s ccache /usr/local/bin/clang
  ln -s ccache /usr/local/bin/clang++
```